File: JULY-2025/14-07-2025/p1290.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head)
    {
        int out = 0;
        int count = 0;
        ListNode og = head;
        // if (head == null)
        //     return 0;
        while(head.next != null)
        {
            count++;
            head = head.next;
        }    

        while(og != null)
        {
            if(og.val == 1)
            {
                out += Math.pow(2,count);
            }
            count--;
            og = og.next;
        }
        return out;
    }
}

File: JULY-2025/14-07-2025/p1290.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        string str="";
        ListNode*curr = head;
        if(!head || !head->next)
        return head->val;
        while(curr){
            str += (curr->val+'0');
            curr = curr->next;
        }
         int len = str.length();
        int num=0;
        int pow = 1;
        for(int i=len-1;i>=0;i--){
            if(str[i]=='1')
            { num += pow;}
            pow *=2;
        }
        return num;
    }
};